activerecord/test/models/category.rb

```diff
@@ -44,4 +44,9 @@ def self.locked
 end
 
 class SpecialCategory < Category
+  after_initialize :format_name
+
+  def format_name
+    self.special_name = name&.upcase&.gsub(/[^A-Z]/, "").presence&.slice(0..3) rescue nil
```

Comment:
We only need to verify that the callback runs with the attributes set. Could this just be `self.special_name = name.upcase`?

Reply:
Yeah, a more specific transformation just seemed like a more accurate test than a simple capitalization test that the transformation was definitely being made, and would still be accurately testable apply if `#name` already was formatted in all caps

Reply:
updated for a simpler check with a directly created record to aviod the need for extra insurance